File: C/pascal triangle.cpp
#include<stdio.h>
int main()
{
    int i,j,n,fact1=1,fact2=1,fact3=1,r,k,l;

        for(n=0;n<=5;n++)
        {
            for(l=5;l>=n;l--){
                printf(" ");
            }
            for(r=0;r<=n;r++)
            {
                for(i=1;i<=n;i++){
                    fact1=fact1*i;
                }
                for(i=1;i<=r;i++)
                {
                    fact2=fact2*i;
                }
                for(i=1;i<=n-r;i++){
                    fact3=fact3*i;
                }
                printf(" %d",fact1/(fact2*fact3));
                fact1=1;
                fact2=1;
                fact3=1;
        
            }
            printf("\n");
        }
}

File: Python/pascal_triangle.py
nl=int(input("enter number of lines"))
fact1=1
fact2=1
fact3=1
for n in range(0,(nl+1),1):
    for l in range(nl,(n-1),-1):
        print(" ",end="")
    for r in range(0,(n+1),1):
        for i in range(1,(n+1),1):
            fact1=fact1*i
        for i in range(1,(r+1),1):
            fact2=fact2*i
        for i in range(1,(n-r+1),1):
            fact3=fact3*i
        result=int(fact1/(fact2*fact3))
        print(" ",result,end="")
        fact1=1
        fact2=1
        fact3=1
    print()


